zipline/finance/slippage.py

```diff
@@ -234,21 +266,224 @@ def process_order(self, data, order):
+            simulated_impact = price * NO_DATA_VOLATILITY_SLIPPAGE_IMPACT
+        else:
+            simulated_impact = self.get_simulated_impact(
+                order=order,
+                current_price=price,
+                current_volume=volume,
+                txn_volume=txn_volume,
+                mean_volume=mean_volume,
+                volatility=volatility,
+            )
+
+        impacted_price = \
+            price + math.copysign(simulated_impact, order.direction)
+
+        if fill_price_worse_than_limit_price(impacted_price, order):
+            return None, None
+
+        return impacted_price, math.copysign(txn_volume, order.direction)
+
+    def _get_window_data(self, data, asset, window_length):
+        """
+        Internal utility method to return the trailing mean volume over the
+        past 'window_length' days, and volatility of close prices for a
+        specific asset.
+
+        Parameters
+        ----------
+        data : The BarData from which to fetch the daily windows.
+        asset : The Asset whose data we are fetching.
+        window_length : Number of days of history used to calculate the mean
+            volume and close price volatility.
+
+        Returns
+        -------
+        (mean volume, volatility)
+        """
+        try:
+            values = self._window_data_cache.get(asset, data.current_session)
+        except KeyError:
+            # Add a day because we want 'window_length' complete days,
+            # excluding the current day.
+            volume_history = data.history(
+                asset, 'volume', window_length + 1, '1d',
+            )
+            close_history = data.history(
+                asset, 'close', window_length + 1, '1d',
+            )
+            # Exclude the first value of the percent change array because it is
+            # always just NaN.
+            close_volatility = close_history[:-1].pct_change()[1:].std(
+                skipna=False,
+            )
+            values = {
+                'volume': volume_history[:-1].mean(),
+                'close': close_volatility * SQRT_252,
+            }
+            self._window_data_cache.set(asset, values, data.current_session)
+
+        # from nose.tools import set_trace; set_trace()
+        return values['volume'], values['close']
+
+
+class VolatilityVolumeShare(MarketImpactBase, FutureSlippageModel):
+    """
+    Model slippage for futures contracts according to the following formula:
+
+        new_price = price + (price * MI / 10000),
+
+    where 'MI' is market impact, which is defined as:
+
+        MI = eta * sigma * sqrt(psi)
+
+    Eta is a constant which varies by root symbol.
+    Sigma is 20-day annualized volatility.
+    Psi is the volume traded in the given bar divided by 20-day ADV.
+    """
+    allowed_asset_types = (Future,)
+
+    def __init__(self, volume_limit, eta=ROOT_SYMBOL_TO_ETA):
+        super(VolatilityVolumeShare, self).__init__()
```

Comment:
might be useful to have documentation on this method explaining the types of the parameters, especially `eta` - took me a while to figure out what the allowed types were for it.